nodes/nomos-node/node/src/api/handlers.rs

```diff
@@ -644,7 +651,7 @@ where
         SamplingMempoolAdapter,
         SamplingStorage,
         RuntimeServiceId,
-    >(&handle, blob_id))
+    >(&handle, req.blob_id, req.session))
```

Comment:
Is this in this pr because of testing?

Reply:
@danielSanchezQ  No. This is the consequence of the sampling now requiring the session so we can compare it with current session. This GetCommitments call remains unsolved, where system will return an MismatchError to the user in the edge-case of the wrong session. 
My plan is to open the next PR with the fallback to historical in the commitments case as well. In order to do this I need to expand historic sampling behavior, as currently it is fetching everything together (commitments and shares), and in this case we would only need to fetch commitments.

Does that makes sense?